ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/button-behavior/button-behavior.tsx

```diff
@@ -0,0 +1,86 @@
+/**
```

Comment:
❓ this doesn't seem like a presentational component, maybe you can split the business logic into a container component

Reply:
Why so?